org.mwc.debrief.pepys/src/main/java/org/mwc/debrief/pepys/model/ModelConfiguration.java

```diff
@@ -314,13 +319,33 @@ public ALGORITHM_TYPE getAlgorithmType() {
 			if (databaseConnection instanceof SqliteDatabaseConnection) {
 				return ALGORITHM_TYPE.LEGACY;
 			} else if (databaseConnection instanceof PostgresDatabaseConnection) {
-				return ALGORITHM_TYPE.FAST_MODE;
+				return ALGORITHM_TYPE.FAST_MODE_STORED_PROC;
 			}
 		}
 		// LEGACY AS DEFAULT.
 		return ALGORITHM_TYPE.LEGACY;
 	}
 
+	@Override
+	public String getCommentQuery(final ALGORITHM_TYPE algorithType) {
+		if (getAlgorithmType() == ALGORITHM_TYPE.FAST_MODE) {
```

Comment:
@saulhidalgoaular - we're receiving the algorithm type as a parameter, but also retrieving it here.

I guess we should either delete the parameter, or use it instead of `getAlgorithmType()`

Reply:
I think a Switch statement would give better type checking in these :
```
	@Override
	public String getCommentQuery(final QUERY_STRATEGY algorithmType) {
		switch (algorithmType) {
		case FAST_MODE:
			return "/comments.sql";
		case FAST_MODE_STORED_PROC:
			return "/commentsProc.sql";
		default:
			Application.logError2(ToolParent.ERROR, "Unexpected query strategy" + algorithmType, null);
			return null;
		}
	}

```

Reply:
> @saulhidalgoaular - we're receiving the algorithm type as a parameter, but also retrieving it here.
> 
> I guess we should either delete the parameter, or use it instead of `getAlgorithmType()`

oh, you are absolutely right. I have it twice.